File: Bezrukov Vlad/ForStudy/app/src/main/java/com/example/forstudy/MainActivity.java
package com.example.forstudy;

import android.os.Bundle;

import android.content.Intent;
import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;
import android.view.View;
import android.widget.TextView;
import android.widget.Button;
import android.widget.EditText;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        EdgeToEdge.enable(this);
        setContentView(R.layout.activity_main);
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main), (v, insets) -> {
            Insets systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars());
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom);
            return insets;
        });

        TextView mainTextView;

        mainTextView = findViewById(R.id.main_textview);
        mainTextView.setText("Set in Java!");

        Button startButton = findViewById(R.id.button);

        startButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(MainActivity.this, SecondActivity.class);
                EditText group = findViewById(R.id.editText);
                intent.putExtra("Group", group.getText().toString());
                startActivity(intent);
            }
        });
    }
}

File: Roma/WelcomeApp/app/src/main/java/com/example/welcomeapp/MainActivity.kt
package com.example.welcomeapp

import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.ImageView
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val str = "Прикол2"
        val str2 = "Прикол"

        // Найти элементы интерфейса
        val startButton = findViewById<Button>(R.id.start_button)
        val welcomeText = findViewById<TextView>(R.id.welcome_text)
        val welcomeImage = findViewById<ImageView>(R.id.welcome_image)
        val groupText = findViewById<TextView>(R.id.group_text)
        val textPrikol = findViewById<TextView>(R.id.main_textview)

        // Установить видимость текста и изображения невидимой
        welcomeText.visibility = View.GONE
        welcomeImage.visibility = View.GONE

        // Обработчик нажатия кнопки
        startButton.setOnClickListener {
            if (welcomeText.visibility == View.VISIBLE) {
                welcomeText.visibility = View.GONE
                welcomeImage.visibility = View.GONE
                groupText.visibility = View.GONE
                textPrikol.text = str2
            } else {
                welcomeText.visibility = View.VISIBLE
                welcomeImage.visibility = View.VISIBLE
                groupText.visibility = View.VISIBLE
                textPrikol.text = str
            }
        }

    }
}